### ISSUE TITLE
Required nullables

### ISSUE THREAD

**Issue by jamamp** [Jman012/FloatplaneAPI#1]
Previously, if a property could be `null` in a JSON response from Floatplane, I modeled this as a property in the JSON Schema without that property being present in the `"required": [...]` section of the schema. For strict languages like Swift, this resulted in a Swift Optional (either the value or `null`/`nil`) and worked out okay.

In TypeScript generation, it turns out this was interpreted, probably more correctly, as `string | undefined`. We want this to be `string | null` instead, because the key does appear in the JSON payload but the value is `null` instead of a value.

I need to go through and find all other instances where I've marked these incorrectly as non-required instead of required+nullable and change this. I'll write a quick script to find instances of this in the schema json for me and correct them. After that, I need to re-visit the C# unit tests because I remember doing some funky stuff to make it even more strict with regard to present/non-present keys. I might be able to remove that hacky code and rely more on this method to make it strict.

**Comment by jamamp** [comment]
More findings:

Confirmed that "required" just makes it appear as a key/property in the object, not necessarily that it must have a value vs being null [^1].

The conventional way for JSON Schema to represent a property as being nullable is something like `{ "type": [ "string", "null" ]}` [^2]. However, OpenAPI Specification breaks this convention, by disallowing multiple types being specified in the array [^3] (see Data Types), and instead opting for a new field "nullable" (`{ "type": "string", "nullable": true }`) [^3] (see Schema Object).

[^1]: https://json-schema.org/understanding-json-schema/reference/object.html#required-properties
[^2]: https://json-schema.org/understanding-json-schema/reference/null.html
[^3]: https://swagger.io/specification/

**Comment by jamamp** [comment]
I think I have all of the schema fixed now.

For the unit tests, I removed the `RequiredObjectPropertiesContractResolver` and the `ErrorModelSerializerSettings`. For the former, the `[DataMember("...", IsRequired = true, ...)]` handles this by making the property key required to be present. For the latter, I have the `ErrorModel` properly defined such that `.message` and `.data` are allowed to be undefined and the removal of the former shouldn't interfere with that now.

However, C# with Newtonsoft.Json only handles classes, not structs. So if I create a control test for, e.g., `ContentPostV3Model` which is missing the required `.metadata` object (null, not undefined), it isn't actually throwing an error. If I restored `RequiredObjectPropertiesContractResolver` with `Required.Always` (as opposed to `Required.AllowNull`) then it'll begin throwing errors. But this shouldn't be a blanket setting. 

The code generator uses `System.Runtime.Serialization.DataMember` instead of `Newtonsoft.Json.JsonPropertyAttribute` and so it doesn't specify the more concise level of required-ness that I need for more strict tests. I'll look into this more.
